pymclevel/leveldbpocket.py

```diff
@@ -414,25 +404,34 @@ def _readChunk(self, cx, cz, world, readOptions=None):
                     biomes.shape = (16 ,16)
                     chunk.Biomes = biomes
                 for i in range(16):
-                    r = self._readSubChunk_1plus(cx, cz, i, rop, key)
-                    if type(r) == tuple:
-                        tr, te, en = r
-                        chunk.add_data(terrain=tr, tile_entities=te, entities=en, subchunk=i)
+                    terrain = self._readSubChunk_1plus(cx, cz, i, rop, key)
+                    chunk.add_terrain(terrain=terrain, subchunk=i)
+
+                try:
+                    tile_entities = db.Get(rop, key + "\x31")
+                except RuntimeError:
+                    tile_entities = None
+                try:
+                    entities = db.Get(rop, key + "\x32")
+                except RuntimeError:
+                    entities = None
+                chunk.add_data(tile_entities=tile_entities, entities=entities)
```

Comment:
I think it makes more sense to have one "add_data" and not separate them to the different types of data